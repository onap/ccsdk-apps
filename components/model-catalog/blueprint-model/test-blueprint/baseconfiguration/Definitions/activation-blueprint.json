{
  "metadata": {
    "template_author": "Brinda Santh Muthuramalingam",
    "author-email": "brindasanth@in.ibm.com",
    "user-groups": "ADMIN, OPERATION",
    "template_name": "baseconfiguration",
    "template_version": "1.0.0",
    "template_tags": "brinda, tosca"
  },
  "imports": [
    {
      "file": "Definitions/data_types.json"
    },
    {
      "file": "Definitions/relationship_types.json"
    },
    {
      "file": "Definitions/artifact_types.json"
    },
    {
      "file": "Definitions/node_types.json"
    },
    {
      "file": "Definitions/policy_types.json"
    }
  ],
  "topology_template": {
    "inputs": {
      "request-id": {
        "required": true,
        "type": "string"
      },
      "action-name": {
        "required": true,
        "type": "string"
      },
      "scope-type": {
        "required": true,
        "type": "string"
      },
      "hostname": {
        "required": true,
        "type": "string"
      }
    },
    "node_templates": {
      "resource-assignment-process": {
        "type": "dg-generic",
        "properties": {
          "content": {
            "get_artifact": [
              "SELF",
              "dg-resource-assignment-process"
            ]
          },
          "dependency-node-templates": [
            "resource-assignment"
          ]
        },
        "artifacts": {
          "dg-resource-assignment-process": {
            "type": "artifact-directed-graph",
            "file": "Plans/CONFIG_ResourceAssignment_1.0.0.xml"
          }
        }
      },
      "activate-process": {
        "type": "dg-generic",
        "properties": {
          "content": {
            "get_artifact": [
              "SELF",
              "dg-activate-process"
            ]
          },
          "dependency-node-templates": [
            "activate-jython"
          ]
        },
        "artifacts": {
          "dg-activate-process": {
            "type": "artifact-directed-graph",
            "file": "Plans/CONFIG_ActivateNetconf_1.0.0.xml"
          }
        }
      },
      "assign-activate-process": {
        "type": "dg-generic",
        "properties": {
          "content": {
            "get_artifact": [
              "SELF",
              "dg-assign-activate-process"
            ]
          },
          "dependency-node-templates": [
            "resource-assignment",
            "activate-jython"
          ]
        },
        "artifacts": {
          "dg-assign-activate-process": {
            "type": "artifact-directed-graph",
            "file": "Plans/CONFIG_AssignActivateNetconf_1.0.0.xml"
          }
        }
      },
      "resource-assignment": {
        "type": "component-resource-resolution",
        "interfaces": {
          "ResourceResolutionComponent": {
            "operations": {
              "process": {
                "inputs": {
                  "action-name": {
                    "get_input": "action-name"
                  },
                  "resource-type": "vnf-type",
                  "request-id": {
                    "get_input": "request-id"
                  },
                  "resource-id": {
                    "get_input": "hostname"
                  },
                  "artifact-prefix-names": [
                    "baseconfig"
                  ]
                },
                "outputs": {
                  "resource-assignment-params": {
                    "get_attribute": [
                      "SELF",
                      "assignment-params"
                    ]
                  },
                  "status": "success"
                }
              }
            }
          }
        },
        "artifacts": {
          "baseconfig-template": {
            "type": "artifact-template-velocity",
            "file": "Templates/baseconfig-template.vtl"
          },
          "baseconfig-mapping": {
            "type": "artifact-mapping-resource",
            "file": "Definitions/baseconfig-mapping.json"
          }
        }
      },
      "resource-assignment-py": {
        "type": "component-resource-resolution",
        "interfaces": {
          "ResourceResolutionComponent": {
            "operations": {
              "process": {
                "implementation": {
                  "primary": "component-script"
                },
                "inputs": {
                  "action-name": {
                    "get_input": "action-name"
                  }
                },
                "outputs": {
                  "resource-assignment-params": "",
                  "status": ""
                }
              }
            }
          }
        },
        "artifacts": {
          "component-script": {
            "type": "artifact-script-jython",
            "file": "Scripts/python/SamplePythonComponentNode.py"
          }
        }
      },
      "activate-jython": {
        "type": "component-jython-executor",
        "interfaces": {
          "ComponentJythonExecutor": {
            "operations": {
              "process": {
                "implementation": {
                  "primary": "component-script"
                },
                "inputs": {
                  "instance-dependencies": [
                    "json-parser-service",
                    "netconf-rpc-service"
                  ]
                },
                "outputs": {
                  "response-data": "",
                  "status": ""
                }
              }
            }
          }
        },
        "artifacts": {
          "component-script": {
            "type": "artifact-script-jython",
            "file": "Scripts/python/SamplePythonComponentNode.py"
          }
        }
      },
      "activate-netconf": {
        "type": "component-netconf-executor",
        "interfaces": {
          "NetconfExecutorComponent": {
            "operations": {
              "process": {
                "implementation": {
                  "primary": "component-script"
                },
                "inputs": {
                  "instance-dependencies": [
                    "json-parser-service",
                    "netconf-rpc-service"
                  ]
                },
                "outputs": {
                  "response-data": "",
                  "status": ""
                }
              }
            }
          }
        },
        "requirements": {
          "netconf-connection": {
            "capability": "netconf",
            "node": "sample-netconf-device",
            "relationship": "tosca.relationships.ConnectsTo"
          }
        },
        "artifacts": {
          "component-script": {
            "type": "artifact-script-jython",
            "file": "Scripts/python/DefaultGetNetConfig.py"
          }
        }
      },
      "sample-netconf-device": {
        "type": "vnf-netconf-device",
        "capabilities": {
          "netconf": {
            "properties": {
              "login-key": "sample-key",
              "login-account": "sample-account",
              "target-ip-address": "localhost",
              "port-number": 830,
              "connection-time-out": 30
            }
          }
        }
      }
    },
    "workflows": {
      "resource-assignment": {
        "inputs": {
          "resource-assignment-properties": {
            "required": true,
            "type": "dt-resource-assignment-properties"
          }
        },
        "steps": {
          "call-resource-assignment": {
            "description": "Resource Assignment Workflow",
            "target": "resource-assignment-process",
            "activities": [
              {
                "call_operation": "CONFIG.ResourceAssignment"
              }
            ]
          }
        }
      },
      "activate": {
        "inputs": {
          "request-id": {
            "required": true,
            "type": "string"
          },
          "action-name": {
            "required": true,
            "type": "string"
          },
          "scope-type": {
            "required": true,
            "type": "string"
          },
          "hostname": {
            "required": true,
            "type": "string"
          }
        },
        "steps": {
          "activate-process": {
            "description": "Netconf Activation Workflow",
            "target": "activate-process",
            "activities": [
              {
                "call_operation": "CONFIG.ActivateProcess"
              }
            ]
          }
        }
      },
      "assign-activate": {
        "inputs": {
          "assign-activate-properties": {
            "required": true,
            "type": "dt-assign-activate-properties"
          }
        },
        "steps": {
          "activate-process": {
            "description": "Resource Assign and Netconf Activation Workflow",
            "target": "assign-activate-process",
            "activities": [
              {
                "call_operation": "CONFIG.AssignActivateProcess"
              }
            ]
          }
        }
      }
    }
  }
}