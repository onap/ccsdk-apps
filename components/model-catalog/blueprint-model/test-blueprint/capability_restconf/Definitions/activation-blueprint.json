<!-- QUESTION
     lines 74-78 and 157-161: if I have to resolve the variables I need based on
     the pnf id, do I need to specify a restconf-connection? Another point, the
     connection is aimed to ODL, not to the PNF, so I am not sure what to fill
     for this model (lines 122-137)
-->
{
  "tosca_definitions_version" : "controller_blueprint_1_0_0",
  "metadata" : {
    "template_author" : "Rodrigo Ottero",
    "author-email" : "rodrigo.ottero@est.tech",
    "user-groups" : "ADMIN, OPERATION",
    "template_name" : "pnf restconf",
    "template_version" : "1.0.0",
    "template_tags" : "pnf, restconf, config, configuration"
  },
  "imports" : [ {
    "file" : "Definitions/data_types.json"
  }, {
    "file" : "Definitions/relationship_types.json"
  }, {
    "file" : "Definitions/artifact_types.json"
  }, {
    "file" : "Definitions/node_types.json"
  }, {
    "file" : "Definitions/policy_types.json"
  } ],
  "topology_template" : {
    "inputs" : {
      "serviceInstanceId" : {
        "required" : true,
        "type" : "string"
      },
      "identifier" : {
        "required" : true,
        "type" : "string"
      }
    },
    "workflows" : {
      "config-assign" : {
        "steps" : {
          "activate-process" : {
            "description" : "Create a configlet",
            "target" : "config-assign-process",
            "activities" : [ {
              "call_operation" : ""
            } ]
          }
        },
        "inputs" : {
          "config-assign-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(config-assign).",
            "required" : true,
            "type" : "dt-config-assign-properties"
          }
        }
      },
      "configure" : {
        "steps" : {
          "activate-process" : {
            "description" : "Send a configlet to the pnf",
            "target" : "configure-process",
            "activities" : [ {
              "call_operation" : ""
            } ]
          }
        },
        "inputs" : {
          "configure-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(configure).",
            "required" : true,
            "type" : "dt-configure-properties"
          }
        }
      }
    },
    "node_templates" : {
      "config-assign-process" : {
        "type" : "dg-generic",
        "properties" : {
          "content" : {
            "get_artifact" : [ "SELF", "dg-config-assign-process" ]
          },
          "dependency-node-templates" : [ "config-assign" ]
        },
        "artifacts" : {
          "dg-config-assign-process" : {
            "type" : "artifact-directed-graph",
            "file" : "Plans/CONFIG_configAssign.xml"
          }
        }
      },
      "config-assign" : {
        "type" : "component-restconf-executor",
        "requirements" : {
          "restconf-connection" : {
            "capability" : "restconf",
            "node" : "sample-restconf-server",
            "relationship" : "tosca.relationships.ConnectsTo"
          }
        },
        "interfaces" : {
          "ComponentRestconfExecutor" : {
            "operations" : {
              "process" : {
                "implementation" : {
                  "primary" : "component-script"
                },
                "inputs" : {
                  "script-type" : "kotlin",
                  "script-class-reference" : "Scripts/kotlin/RestconfAssignConfig.kts",
                  "instance-dependencies" : [ ],
                  "artifact-prefix-names": [
                      "config-assign"
                  ]
                },
                "outputs" : {
                  "response-data" : "",
                  "status" : ""
                }
              }
            }
          }
        },
        "artifacts" : {
          "config-assign-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/config-assign-restconf-configlet-template.vtl"
          },
          "config-assign-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/config-assign-pnf-mapping.json"
          },
          "component-script" : {
            "type" : "artifact-script-kotlin",
            "file" : "Scripts/kotlin/RestconfAssignConfig.py"
          }
        }
      },
      "configure-process" : {
        "type" : "dg-generic",
        "properties" : {
          "content" : {
            "get_artifact" : [ "SELF", "dg-configure-process" ]
          },
          "dependency-node-templates" : [ "configure" ]
        },
        "artifacts" : {
          "dg-config-assign-process" : {
            "type" : "artifact-directed-graph",
            "file" : "Plans/CONFIG_configure.xml"
          }
        }
      },
      "configure" : {
        "type" : "component-restconf-executor",
        "requirements" : {
          "restconf-connection" : {
            "capability" : "restconf",
            "node" : "sample-restconf-server",
            "relationship" : "tosca.relationships.ConnectsTo"
          }
        },
        "interfaces" : {
          "ComponentRestconfExecutor" : {
            "operations" : {
              "process" : {
                "implementation" : {
                  "primary" : "component-script"
                },
                "inputs" : {
                  "script-type" : "kotlin",
                  "script-class-reference" : "Scripts/kotlin/RestconfConfigure.kts",
                  "instance-dependencies" : [ ],
                  "artifact-prefix-names": [
                      "configure-restconf-mount", "configure-restconf-unmount"
                  ]
                },
                "outputs" : {
                  "response-data" : "",
                  "status" : ""
                }
              }
            }
          }
        },
        "artifacts" : {
          "configure-mount-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/configure-restconf-mount-template.vtl"
          },
          "configure-unmount-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/configure-restconf-unmount-template.vtl"
          },
          "configure-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/configure-pnf-mapping.json"
          },
          "component-script" : {
            "type" : "artifact-script-kotlin",
            "file" : "Scripts/kotlin/RestconfConfigure.py"
          }
        }
      },
      "sample-restconf-device" : {
        "type" : "pnf-netconf-device",
        "capabilities" : {
          "restconf" : {
            "properties" : {
              "login-key" : {
                "get_input" : "password"
              },
              "login-account" : {
                "get_input" : "username"
              },
              "target-ip-address" : {
                "get_input" : "ip"
              },
              "port-number" : 830,
              "connection-time-out" : 5
            }
          }
        }
      }
    }
  }
}
