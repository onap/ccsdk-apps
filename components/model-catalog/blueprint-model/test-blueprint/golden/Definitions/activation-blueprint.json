{
  "tosca_definitions_version" : "controller_blueprint_1_0_0",
  "metadata" : {
    "template_author" : "Alexis de TalhouÃ«t",
    "author-email" : "adetalhouet89@gmail.com",
    "user-groups" : "ADMIN, OPERATION",
    "template_name" : "test",
    "template_version" : "1.0.0",
    "template_tags" : "test"
  },
  "imports" : [ {
    "file" : "Definitions/data_types.json"
  }, {
    "file" : "Definitions/relationship_types.json"
  }, {
    "file" : "Definitions/artifact_types.json"
  }, {
    "file" : "Definitions/node_types.json"
  }, {
    "file" : "Definitions/policy_types.json"
  } ],
  "dsl_definitions" : {
    "ipam-1" : {
      "type" : "token-auth",
      "url" : "http://localhost:32778",
      "token" : "Token 0123456789abcdef0123456789abcdef01234567"
    },
    "config-deploy-properties" : {
      "resolution-key" : {
        "get_input" : "resolution-key"
      }
    }
  },
  "topology_template" : {
    "workflows" : {
      "resource-assignment" : {
        "steps" : {
          "resource-assignment-process" : {
            "description" : "Resource Assign Workflow",
            "target" : "resource-assignment-process",
            "activities" : [ {
              "call_operation" : ""
            } ]
          }
        },
        "inputs" : {
          "artifact-name" : {
            "required" : true,
            "type" : "list",
            "entry_schema" : {
              "type" : "string"
            }
          },
          "store-result" : {
            "required" : true,
            "type" : "boolean"
          },
          "resolution-key" : {
            "required" : true,
            "type" : "string"
          },
          "resource-assignment-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(resource-assignment).",
            "required" : true,
            "type" : "dt-resource-assignment-properties"
          }
        }
      },
      "config-deploy" : {
        "steps" : {
          "config-deploy" : {
            "description" : "Resource Assign and Python Netconf Activation Workflow",
            "target" : "config-deploy-process",
            "activities" : [ {
              "call_operation" : ""
            } ]
          }
        },
        "inputs" : {
          "resolution-key" : {
            "required" : false,
            "type" : "string"
          },
          "password" : {
            "required" : true,
            "type" : "string"
          },
          "username" : {
            "required" : true,
            "type" : "string"
          },
          "ip" : {
            "required" : true,
            "type" : "string"
          },
          "config-deploy-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(config-deploy).",
            "required" : true,
            "type" : "dt-config-deploy-properties"
          }
        }
      },
      "rollback" : {
        "steps" : {
          "rollback" : {
            "description" : "Roolback Workflow",
            "target" : "rollback-process",
            "activities" : [ {
              "call_operation" : ""
            } ]
          }
        },
        "inputs" : {
          "resolution-key" : {
            "required" : false,
            "type" : "string"
          },
          "password" : {
            "required" : true,
            "type" : "string"
          },
          "username" : {
            "required" : true,
            "type" : "string"
          },
          "ip" : {
            "required" : true,
            "type" : "string"
          },
          "config-deploy-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(config-deploy).",
            "required" : true,
            "type" : "dt-config-deploy-properties"
          },
          "rollback-properties" : {
            "description" : "Dynamic PropertyDefinition for workflow(rollback).",
            "required" : true,
            "type" : "dt-rollback-properties"
          }
        }
      }
    },
    "node_templates" : {
      "resource-assignment-process" : {
        "type" : "dg-generic",
        "properties" : {
          "content" : {
            "get_artifact" : [ "SELF", "dg-resource-assignment-process" ]
          },
          "dependency-node-templates" : [ "resource-assignment" ]
        },
        "artifacts" : {
          "dg-resource-assignment-process" : {
            "type" : "artifact-directed-graph",
            "file" : "Plans/ResourceAssignment.xml"
          }
        }
      },
      "resource-assignment" : {
        "type" : "component-resource-resolution",
        "interfaces" : {
          "ResourceResolutionComponent" : {
            "operations" : {
              "process" : {
                "inputs" : {
                  "resolution-key" : {
                    "get_input" : "resolution-key"
                  },
                  "store-result" : {
                    "get_input" : "store-result"
                  },
                  "artifact-prefix-names" : {
                    "get_input" : "artifact-name"
                  }
                },
                "outputs" : {
                  "resource-assignment-params" : {
                    "get_attribute" : [ "SELF", "assignment-params" ]
                  },
                  "status" : "success"
                }
              }
            }
          }
        },
        "artifacts" : {
          "hostname-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/hostname-template.vtl"
          },
          "hostname-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/hostname-mapping.json"
          },
          "vf-module-1-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/vf-module-1-template.vtl"
          },
          "vf-module-1-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/vf-module-1-mapping.json"
          }
        }
      },
      "config-deploy-process" : {
        "type" : "dg-generic",
        "properties" : {
          "content" : {
            "get_artifact" : [ "SELF", "dg-config-deploy-process" ]
          },
          "dependency-node-templates" : [ "config-deploy" ]
        },
        "artifacts" : {
          "dg-config-deploy-process" : {
            "type" : "artifact-directed-graph",
            "file" : "Plans/ConfigDeploy.xml"
          }
        }
      },
      "config-deploy" : {
        "type" : "component-netconf-executor",
        "requirements" : {
          "netconf-connection" : {
            "capability" : "netconf",
            "node" : "netconf-device",
            "relationship" : "tosca.relationships.ConnectsTo"
          }
        },
        "interfaces" : {
          "ComponentNetconfExecutor" : {
            "operations" : {
              "process" : {
                "inputs" : {
                  "script-type" : "jython",
                  "script-class-reference" : "Scripts/python/NetconfRpcExample.py",
                  "instance-dependencies" : [ ],
                  "dynamic-properties" : "*config-deploy-properties"
                },
                "outputs" : {
                  "response-data" : "",
                  "status" : "SUCCESS"
                }
              }
            }
          }
        },
        "artifacts" : {
          "vfw-interface-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/vfw-interface-template.vtl"
          },
          "vfw-interface-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/vfw-interface-mapping.json"
          }
        }
      },
      "rollback-process" : {
        "type" : "dg-generic",
        "properties" : {
          "content" : {
            "get_artifact" : [ "SELF", "dg-rollback-process" ]
          },
          "dependency-node-templates" : [ "config-deploy" ]
        },
        "artifacts" : {
          "dg-rollback-process" : {
            "type" : "artifact-directed-graph",
            "file" : "Plans/Rollback.xml"
          }
        }
      },
      "rollback" : {
        "type" : "component-netconf-executor",
        "requirements" : {
          "netconf-connection" : {
            "capability" : "netconf",
            "node" : "netconf-device",
            "relationship" : "tosca.relationships.ConnectsTo"
          }
        },
        "interfaces" : {
          "ComponentNetconfExecutor" : {
            "operations" : {
              "process" : {
                "inputs" : {
                  "script-type" : "jython",
                  "script-class-reference" : "Scripts/python/Rollback.py",
                  "instance-dependencies" : [ ],
                  "dynamic-properties" : "*config-deploy-properties"
                },
                "outputs" : {
                  "response-data" : "",
                  "status" : "SUCCESS"
                }
              }
            }
          }
        },
        "artifacts" : {
          "junos-rollback-RPC-template" : {
            "type" : "artifact-template-velocity",
            "file" : "Templates/junos-rollback-RPC-template.vtl"
          },
          "junos-rollback-RPC-mapping" : {
            "type" : "artifact-mapping-resource",
            "file" : "Templates/junos-rollback-RPC-mapping.json"
          }
        }
      },
      "netconf-device" : {
        "type" : "vnf-netconf-device",
        "capabilities" : {
          "netconf" : {
            "properties" : {
              "login-key" : {
                "get_input" : "password"
              },
              "login-account" : {
                "get_input" : "username"
              },
              "target-ip-address" : {
                "get_input" : "ip"
              },
              "port-number" : 830,
              "connection-time-out" : 5
            }
          }
        }
      }
    }
  }
}